### PR TITLE
Adding association collection and more to association

### PR DESCRIPTION
**Pull request by kieranderson**

### CODE REVIEW

File: src/activities/Association.js [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-import { Actions, Rels } from '../hypermedia-constants';
+import { Actions, Classes, Rels } from '../hypermedia-constants';
 import { performSirenAction } from '../es6/SirenAction';
-
 /**
  * Association class representation of an association.
  */
 
-export class Association extends Entity {
+export class AssociationEntity extends Entity {
```
Comment (ppaskaris-d2l) [inline src/activities/Association.js]: Can you also `export const Association = AssociationEntity;` so it doesn't break? multi-repo madness and all...

File: src/activities/AssociationCollectionEntity.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+export class AssociationCollectionEntity extends Entity {
+
+	getAllAssociations() {
+		if (!this._entity) {
+			return [];
+		}
+
+		const associations = this._entity.getSubEntitiesByClass(
```
Comment (ppaskaris-d2l) [inline src/activities/AssociationCollectionEntity.js]: Strongly consider checking the `rel`

File: src/activities/Associations.js [hunk 3]
```diff
@@ -1,26 +1,19 @@
...
 
-		const singleAssociations = this._entity.getSubEntitiesByClass(
-			Classes.associations.singleAssociation
-		);
-		const singleAssociationsHrefs = singleAssociations.map(
-			a => a.getLinkByRel('self').href
-		);
-
-		return singleAssociationsHrefs;
+		return this._entity.getSubEntitiesByRel('item');
```
Comment (ppaskaris-d2l) [inline src/activities/Associations.js]: Won't this just return vanilla siren entities, and not the wrapper objects? Is that what you want?

Also, now no filtering by single/potential association?

I guess the intent of this method has now changed.

Comment (kieranderson) [reply src/activities/Associations.js]: I had a separate CollectionEntity that was using this but I moved it here to associations

**Pull request merged**